woocommerce/woocommerce-gutenberg-products-block
# Fix wp-cli container permissions issues for e2e docker test environment

In addition to fixing permission issues with the wp-cli container in the docker setup for e2e tests ( see #1670 for original pull that this one is based off), this pull also:

- expands npm scripts for docker commands.
- modifies `docker-compose.yml` so it reads from an `.env` file (allows for easier configuration across repos in future iterations).
- reverts pulling in woocommerce core via composer (done internally in the new `entryscript.js`.
- Adds a custom `Dockerfile` config for the wp-cli container. **Note:** this gets things working well for the blocks repo, however in future iterations I'd like the config to be repo agnostic (which would allow for customizing set volumes and plugins installed via the `.env`.
- Update `.travis.yml` to run e2e tests container (which hopefully works 🤞 )

## To test

If you have pre-existing containers build from the previous e2e setup, you'll want to run `npm run docker:clear-all` to clear out all your docker images and start from scratch.  Then:

-  `npm run docker:up` 
- `npm run tests:e2e` (and verify tests pass).

## Other

- `npm run docker:up:build` will do a rebuild if you tweak any `Dockerfile` or `entryscript.js` 
- `npm run docker:down:reset` will unmount volumes so when you do `npm run docker:up` it re-installs a fresh WP instance.

## Followups

The below ideally would be done as a part of WooCommerce core and the package being worked on there.

- `/bin/docker/wp-cli/entryscript.js` can be tweaked so it receives the list of plugins to install and activate via `.env` variable. That way it can be a bit more repo agnostic and more flexible.
- Once the above is done, we could create a custom image on docker hub for the `wp-cli` container so that it can be setup quicker on travis runs.
- `docker-compose.yml` should read from environment variables (`.env`) for the volumes to mount and where they are mounted in the container. Important for being repo agnostic.
